File: compiler/testdata/go/test_control.go
package main

import (
	"fmt"

	"github.com/redneckbeard/thanos/stdlib"
)

func main() {
	x := 100
	for x > 0 {
		x--
	}
	for x != 50 {
		x++
	}
	y := 0
	for x {
		y++
		if y > 5 {
			break
		}
	}
	for !(y > 100) {
		y++
		if y%2 == 0 {
			continue
		}
		fmt.Println(y)
	}
	var x int
	for _, x = range []int{1, 2, 3, 4} {
		fmt.Println(x)
		if x == 3 {
			break
		}
	}
	var k string
	var v int
	om := stdlib.NewOrderedMap[string, int]()
	om.Set("foo", 1)
	om.Set("bar", 2)
	om.Set("baz", 3)
	om.Set("quux", 4)
	for k, v = range om.All() {
		if k == "foo" || v == 10 {
			continue
		}
	}
}

File: compiler/testdata/ruby/test_control.rb
x = 100

while x > 0 do
  x -= 1
end

until x == 50 do
  x += 1
end

y = 0

while x do
  y += 1
  if y > 5
    break
  end
end

until y > 100
  y += 1
  if y % 2 == 0
    next
  end
  puts y
end

for x in [1, 2, 3, 4] do
  puts x
  break if x == 3
end

for k, v in {foo: 1, bar: 2, baz: 3, quux: 4} do
  next if k == :foo || v == 10
end
